R/pkg/R/client.R

```diff
@@ -60,13 +60,48 @@ generateSparkSubmitArgs <- function(args, sparkHome, jars, sparkSubmitOpts, pack
   combinedArgs
 }
 
+checkJavaVersion <- function() {
+  javaBin <- "java"
+  javaHome <- Sys.getenv("JAVA_HOME")
+  javaReqs <- utils::packageDescription(utils::packageName(), fields=c("SystemRequirements"))
+  sparkJavaVersion <- as.numeric(tail(strsplit(javaReqs, "[(=)]")[[1]], n = 1L))
+  if (javaHome != "") {
+    javaBin <- file.path(javaHome, "bin", javaBin)
```

Comment:
this `bin` - is it applicable to Windows?

Reply:
Yes. If you look at https://github.com/apache/spark/blob/d3c426a5b02abdec49ff45df12a8f11f9e473a88/bin/spark-class2.cmd#L54 we append `bin`

Reply:
ah ok